oeps/oep-0045/decisions/0001-tutor-as-replacement-for-edx-configuration.rst

```diff
@@ -0,0 +1,499 @@
+
+3. Because the master branch of ``edx-platform`` is not supported at the time of writing, one must clone Open edX at the
+   latest supported release *tag*:
+
+   .. code-block::
+
+      cd $WORKSPACE
+      git clone -b open-release/koa.1 git@github.com:edx/edx-platform.git
+
+   .. note::
+
+      Note that branches based off of ``master``, or even ``koa.master``, will not work in production.  As noted
+      `in the documentation <https://docs.tutor.overhang.io/configuration.html#running-a-fork-of-edx-platform>`__, "Tutor
+      will attempt to apply security and bug fix patches that might already be included", and possibly fail.
+
+4. Create a docker-compose override file for the development environment:
+
+   .. code-block::
+
+      vim ${TUTOR_ROOT}/env/dev/docker-compose.override.yml
+
+   And configure it so that your local checkout of ``edx-platform`` is bind-mounted in the correct directory by all
+   relevant containers:
+
+   .. code-block::
+
+      version: "3.7"
+      services:
+        lms:
+          volumes:
+            - ${WORKSPACE}/edx-platform/:/openedx/edx-platform
+        cms:
+          volumes:
+            - ${WORKSPACE}/edx-platform/:/openedx/edx-platform
+        lms-worker:
+          volumes:
+            - ${WORKSPACE}/edx-platform/:/openedx/edx-platform
+        cms-worker:
+          volumes:
+            - ${WORKSPACE}/edx-platform/:/openedx/edx-platform
+
+5. Prepare the ``edx-platform`` repository inside the LMS container:
+
+   .. code-block::
+
+      tutor dev run lms bash
+      pip install -r requirements/edx/development.txt
+      npm install
+      openedx-assets build --env=dev
+      exit
+
+6. You'll need to init the dev environment.  (This is taken care of by ``tutor local quickstart`` for the local
+   production environment, and even though the same databases are used for the development environment, apparently the
+   ``GRANT`` command needs to be run again when MySQL runs on a different container.)
+
+   .. code-block::
+
+      tutor dev init
+
+7. And finally, to start the LMS, CMS, and workers in the background:
+
+   .. code-block::
+
+      tutor dev start -d
+
+Code and asset reloading
+````````````````````````
+
+When ``edx-platform`` is mounted as described above, changes made to python files are reloaded immediately.
+
+Template and asset changes are also available immediately, but a browser refresh and/or a Redis cache flush may be
+necessary:
+
+.. code-block::
+
+   tutor dev exec redis bash
+   redis-cli -h redis FLUSHALL
+
+Debugging and testing
+`````````````````````
+
+It is possible to `run an interactive debugger <https://docs.tutor.overhang.io/dev.html#debug-edx-platform>`__, as well
+as `running unit tests locally <https://docs.tutor.overhang.io/dev.html#running-edx-platform-unit-tests>`__.
+
+.. note::
+
+   As noted in the Tutor documentation and elsewhere in this ADR, some ``edx-platform`` tests are currently failing.
```

Comment:
35 tests out of 17k are failing -- and I believe the same pytest commands also fail in the Devstack. I think this sentence does not accurately reflect how close we are to getting tests to run on Tutor. For the overwhelming majority of applications, it is already possible to run tests on Tutor.

Reply:
@regisb, fair enough.  I'll replace "some" with "a few", and the exact number out of the total.